orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -0,0 +1,162 @@
+        username: str = None,
+        password: str = None
+    ):
+        username_password_auth_available = username and password
+
+        if not client_secret_key and not username_password_auth_available:
+            raise InvalidConnectionException(
+                ["client_secret_key", "username", "password"],
+                message="Must have either client_secret_key or username and password"
+            )
+
+        self.client_connection = orodha_keycloak.connections.client.create_client_connection(
+            server_url=server_url,
+            realm_name=realm_name,
+            client_id=client_id,
+            client_secret_key=client_secret_key
+        )
+        if client_secret_key:
+            self.admin_connection = orodha_keycloak.connections.admin.create_admin_connection(
+                server_url=server_url,
+                realm_name=realm_name,
+                client_id=client_id,
+                client_secret_key=client_secret_key,
+            )
+        else:
+            self.admin_connection = orodha_keycloak.connections.admin.create_admin_connection(
+                server_url=server_url,
+                realm_name=realm_name,
+                username=username,
+                password=password
+            )
+
+    def add_user(
+            self,
+            email: str,
+            username: str,
+            firstName: str,
+            lastName: str,
+            password: str
+    ):
+        """
+        Adds a user to keycloak with a password.
+
+        Args:
+                email(str): The email of the new user.
+                username(str): The username of the new user.
+                firstName(str): The first name of the new user.
+                lastName(str): The last name of the new user.
+                password(str): The password of the new user.
+
+        Returns:
+            new_user: The new user info genereated by the keycloak server.
+
+        """
+        new_user = self.admin_connection.create_user(
+            {
+                "email": email,
+                "username": username,
+                "enabled": True,
+                "firstName": firstName,
+                "lastName": lastName,
+                "credentials": [
+                    {
+                        "value": password,
+                        "type": "password",
+                    }
+                ],
+            },
+            exist_ok=False,
+        )
+
+        return new_user
+
+    def delete_user(self, user_id):
+        """
+        Deletes a keycloak user with a given user_id.
+
+        Args:
+            user_id(str): The user id of the user to be deleted.
+
+        Returns:
+            response: The response from the keycloak server with info about the deletion.
+        """
+        response = self.admin_connection.delete_user(user_id=user_id)
+
+        return response
+
+    def get_user(self, token: dict = None, user_id: str = None):
```

Comment:
@Edison-Stuart The type hint is incorrect here; `token` is a string, but it's labeled as a dict